hibernate-core/src/main/java/org/hibernate/type/BlobType.java

```diff
@@ -36,7 +40,23 @@ protected boolean registerUnderJavaType() {
 
 	@Override
 	protected Blob getReplacement(Blob original, Blob target, SharedSessionContractImplementor session) {
+		if ( target == null ) {
+			return copyOriginalBlob( (Blob) original, session );
```

Comment:
This is OK for `Dialect#NEW_LOCATOR_LOB_MERGE_STRATEGY` (by default, returned by `Dialect#getLobMergeStrategy), but not for `LEGACY_LOB_MERGE_STRATEGY` or `STREAM_XFER_LOB_MERGE_STRATEGY`. It's unknown if this would work for a custom `LobMergeStrategy`.

Reply:
Actually, it should be delegated to the `LobMergeStrategy`.

Reply:
@gsmet, was there a reason why you didn't just delegate to the `LobMergeStrategy`?